test/TokenRequestTest.js

```diff
@@ -1,51 +1,304 @@
-            const actualVault = await tokenRequest.vault()
+      const actualTRBalance = (await getBalance(tokenRequest.address)).valueOf()
+      const actualNextTokenRequestId = await tokenRequest.nextTokenRequestId()
 
-            assert.strictEqual(actualTokenManager, tokenManager.address)
-            assert.strictEqual(actualVault, vault)
-        })
+      assert.equal(actualTRBalance, expectedTRBalance)
+      assert.equal(actualNextTokenRequestId, expectedNextTokenRequestId)
     })
 
-})
\ No newline at end of file
+    it('should not create a new request for 0 Ether', async () => {
+      await assertRevert(
+        tokenRequest.createTokenRequest(ZERO_ADDRESS, 0, 1, {
+          value: 0,
+        }),
+        'TOKEN_REQUEST_NO_AMOUNT'
+      )
+    })
+
+    it('should not create a new request with different _depositAmount and value', async () => {
+      await assertRevert(
+        tokenRequest.createTokenRequest(ZERO_ADDRESS, 100, 1, {
+          value: 50,
+        }),
+        'TOKEN_REQUEST_ETH_VALUE_MISMATCH'
+      )
+    })
+
+    it('creates a new token request on exchange for TokenMock', async () => {
+      const expectedTRBalance = ROOT_TOKEN_AMOUNT
+      const expectedNextTokenRequestId = 1
+
+      await mockErc20.approve(tokenRequest.address, ROOT_TOKEN_AMOUNT, {
+        from: rootAccount,
+      })
+
+      await tokenRequest.createTokenRequest(mockErc20.address, ROOT_TOKEN_AMOUNT, 300)
+
+      const actualTRBalance = await mockErc20.balanceOf(tokenRequest.address)
+
+      const actualNextTokenRequestId = await tokenRequest.nextTokenRequestId()
+
+      assert.equal(actualTRBalance, expectedTRBalance)
+      assert.equal(actualNextTokenRequestId, expectedNextTokenRequestId)
+    })
+
+    it('should not create a new request without token apporove', async () => {
+      await assertRevert(
+        tokenRequest.createTokenRequest(mockErc20.address, 100, 1),
+        'TOKEN_REQUEST_TOKEN_TRANSFER_REVERTED'
+      )
+    })
+  })
+
+  describe('finaliseTokenRequest(uint256 _tokenRequestId)', () => {
+    let script, forwarderMock, forwarderMockBase
+    beforeEach(async () => {
+      forwarderMockBase = await ForwarderMock.new()
+      const newForwarderMockReceipt = await daoDeployment.kernel.newAppInstance(
+        '0x9876',
+        forwarderMockBase.address,
+        '0x',
+        false,
+        { from: rootAccount }
+      )
+      forwarderMock = await ForwarderMock.at(deployedContract(newForwarderMockReceipt))
+
+      await forwarderMock.initialize()
+
+      await daoDeployment.acl.createPermission(tokenRequest.address, tokenManager.address, MINT_ROLE, rootAccount)
+      await daoDeployment.acl.createPermission(
+        forwarderMock.address,
+        tokenRequest.address,
+        FINALISE_TOKEN_REQUEST_ROLE,
+        rootAccount
+      )
+    })
+
+    it('finalise token request (ERC20)', async () => {
+      const expectedUserMiniMeBalance = 300
+      const expectedVaultBalance = 100
```

Comment:
If the idea is to use `ROOT_TOKEN_AMOUNT` we could use it here instead of the value 100